src/Microsoft.AspNet.Razor/Parser/HtmlMarkupParser.cs

```diff
@@ -134,12 +134,31 @@ protected void SkipToAndParseCode(Func<HtmlSymbol, bool> condition)
                 {
                     if (last != null)
                     {
+                        if (startOfLine && last.Type == HtmlSymbolType.WhiteSpace)
```

Comment:
Should add a comment explaining this one :smile:. This parser loop can be confusing.